src/main/kotlin/uk/gov/gdx/datashare/helpers/TimeLimitedRepeater.kt

```diff
@@ -0,0 +1,19 @@
+package uk.gov.gdx.datashare.helpers
+
+import java.time.LocalDateTime
+
+class TimeLimitedRepeater {
+  companion object {
+
+    private const val defaultTimeLimit: Long = 45
+    fun <T> repeat(items: List<T>, repeatableFunction: (item: T) -> Unit, timeLimitInSeconds: Long? = null) {
+      val startTime = LocalDateTime.now()
+      items.forEach {
+        repeatableFunction(it)
+        if (LocalDateTime.now() > startTime.plusSeconds(timeLimitInSeconds ?: defaultTimeLimit)) {
+          return
+        }
+      }
+    }
+  }
+}
```

Comment:
Is there a reason we've got this in a class?
```suggestion
fun <T> repeatForLimitedTime(items: Iterable<T>, action: (item: T) -> Unit, timeLimitInSeconds: Long = 45) {
  val startTime = LocalDateTime.now()
  items.forEach {
    action(it)
    if (LocalDateTime.now() > startTime.plusSeconds(timeLimitInSeconds)) {
      return
    }
  }
} 
```